src/main/java/com/kylenicholls/stash/parameterizedbuilds/ciserver/Jenkins.java

```diff
@@ -116,7 +116,8 @@ public Server getJenkinsServer() {
 			// legacy settings
 			String[] serverProps = settingObj.toString().split(";");
 			boolean altUrl = serverProps.length > 3 && "true".equals(serverProps[3]) ? true : false;
-			return new Server(serverProps[0], serverProps[1], serverProps[2], altUrl);
+			boolean csrfEnabled = (serverProps.length > 4 && "true".equals(serverProps[4])) ? true : false;
```

Comment:
I'd prefer that the default if `serverProps.length <= 4` would be `true`. That way if anyone was using the csrf protection the old way and upgrades to a version with this change, their jobs don't stop triggering.

Reply:
there are multiple places to set the default value, I'll have a look at that